<launch>
    <arg name="bag_file" default="/media/eugeniu/T71/2025-07-09_Evo/1014/2025-07-09-11-54-55.bag"/>
    
    <!--  
    <node pkg="rosbag" type="play" name="rosbag_play" args=" clock -r .6 -s 0. $(arg bag_file) " output="log"/>
    -->

    <arg name="rviz" default="true" />

    <rosparam command="load" file="$(find s_mls)/config/hesai_drone.yaml" />

    <param name="max_iteration" type="int" value="3" /> <!--3-->
    <param name="Bag" type="string" value="$(arg bag_file)" />
    
    <param name="filter_size_surf" type="double" value=".1" />
    <param name="filter_size_map" type="double" value=".1" />
    <param name="cube_side_length" type="double" value="120" /> <!--1000-->

    <!-- 
    <node pkg="s_mls" type="mls_mapping" name="mls_mapping" output="screen" /> 
    -->
    <!-- 
    <node pkg="s_mls" type="georeference" name="georeference" output="screen" /> 
    -->
    <node pkg="s_mls" type="test_fusion" name="test_fusion" output="screen" /> 
    


    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find s_mls)/rviz_cfg/visualize.rviz" /> 
    </group>

    
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server" ns="MLS" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="MLS" />
        <param name="/trajectory_update_rate" value="15.0" />
        <param name="/trajectory_publish_rate" value="15.0" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_GPSFix" ns="GPSFix" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="GPSFix" />
        <param name="/trajectory_update_rate" value="15.0" />
        <param name="/trajectory_publish_rate" value="15.0" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_PPK_GNSS" ns="PPK_GNSS" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="PPK_GNSS" />
        <param name="/trajectory_update_rate" value="50.0" />
        <param name="/trajectory_publish_rate" value="50.0" />
    </node>
</launch>
